### ISSUE TITLE
Is it normal being there 86% trainable parameters while training?

### ISSUE THREAD

**Issue by feiyangsuo** [lllyasviel/ControlNet#321]
First of all, thanks for the great work of yours!

I'm using code from tutorial_train.py to train controlnet with fill50k following https://github.com/lllyasviel/ControlNet/blob/main/docs/train.md
While everything seems OK, I found these param summaries indicating there are about **86%** trainable parameters in the network:

```
  | Name              | Type               | Params
---------------------------------------------------------
0 | model             | DiffusionWrapper   | 859 M 
1 | first_stage_model | AutoencoderKL      | 83.7 M
2 | cond_stage_model  | FrozenCLIPEmbedder | 123 M 
3 | control_model     | ControlNet         | 361 M 
---------------------------------------------------------
1.2 B     Trainable params
206 M     Non-trainable params
1.4 B     Total params
5,710.058 Total estimated model params size (MB)
```
As I've set `sd_locked = True`, to my knowledge the trainable parameters should be this part:
![image](https://user-images.githubusercontent.com/47778268/228787147-1e2ce28a-6195-4688-9b25-e418c4546cbd.png)
These parameters doesn't seem to occupy 86% of the network. 
I'm worrying about having something messed up and the whole diffusion network performance may broke.

full stack:
```
No module 'xformers'. Proceeding without it.
ControlLDM: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
making attention of type 'vanilla' with 512 in_channels
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
making attention of type 'vanilla' with 512 in_channels
Loaded model config from [./models/cldm_v15.yaml]
Loaded state_dict from [./weights/v1-5-pruned.controlnet.ckpt]
GPU available: True, used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
/home/pai/lib/python3.9/site-packages/pytorch_lightning/trainer/configuration_validator.py:118: UserWarning: You defined a `validation_step` but have no `val_dataloader`. Skipping val loop.
  rank_zero_warn("You defined a `validation_step` but have no `val_dataloader`. Skipping val loop.")
/home/pai/lib/python3.9/site-packages/pytorch_lightning/trainer/configuration_validator.py:280: LightningDeprecationWarning: Base `LightningModule.on_train_batch_start` hook signature has changed in v1.5. The `dataloader_idx` argument will be removed in v1.7.
  rank_zero_deprecation(
/home/pai/lib/python3.9/site-packages/pytorch_lightning/trainer/configuration_validator.py:287: LightningDeprecationWarning: Base `Callback.on_train_batch_end` hook signature has changed in v1.5. The `dataloader_idx` argument will be removed in v1.7.
  rank_zero_deprecation(
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]

  | Name              | Type               | Params
---------------------------------------------------------
0 | model             | DiffusionWrapper   | 859 M 
1 | first_stage_model | AutoencoderKL      | 83.7 M
2 | cond_stage_model  | FrozenCLIPEmbedder | 123 M 
3 | control_model     | ControlNet         | 361 M 
---------------------------------------------------------
1.2 B     Trainable params
206 M     Non-trainable params
1.4 B     Total params
5,710.058 Total estimated model params size (MB)
Epoch 0:   0%|                                                                                                  | 0/50000 [00:00<?, ?it/s]/home/pai/lib/python3.9/site-packages/pytorch_lightning/utilities/data.py:56: UserWarning: Trying to infer the `batch_size` from an ambiguous collection. The batch size we found is 1. To avoid any miscalculations, use `self.log(..., batch_size=batch_size)`.
  warning_cache.warn(
Data shape for DDIM sampling is (1, 4, 64, 64), eta 0.0
Running DDIM Sampling with 50 timesteps
DDIM Sampler: 100%|███████████████████████████████████████████████████████████████████████████████████████| 50/50 [00:11<00:00,  4.46it/s]
Epoch 0:   0%| | 127/50000 [01:20<8:44:24,  1.59it/s, loss=0.0104, v_num=2, train/loss_simple_step=0.0163, train/loss_vlb_step=0.000106, t
```

**Comment by lllyasviel** [comment]
"Trainable" means params can generate gradients.
"Trainable" does not mean params are updated.
